zebra-state/src/service/finalized_state/disk_format/block.rs

```diff
@@ -91,9 +91,7 @@ impl TransactionIndex {
 
     /// Returns this index as a `usize`
     pub fn as_usize(&self) -> usize {
-        self.0
-            .try_into()
-            .expect("the maximum valid index fits in usize")
+        self.0.into()
     }
 
     /// Creates a transaction index from a `u64`.
```

Comment:
Here are some possible reasons:
- integer constants can't be used to convert into the type, because there are multiple possible From methods for a generic integer
- it's hard to see what integer types are actually being converted in reviews, and sometimes that is important due to their byte size or potential overflow
- the code is outdated and could be cleaned up

In high-level code using From is better, but in the database sometimes it's important to know how big things are. I'm not sure if that applies here.